app/grandchallenge/algorithms/views.py

```diff
@@ -386,43 +389,54 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
-class AlgorithmResultsList(PermissionListMixin, ListView):
+class AlgorithmResultsList(PermissionListMixin, PaginatedTableListView):
     model = Result
     permission_required = (
         f"{Result._meta.app_label}.view_{Result._meta.model_name}"
     )
+    row_template = "algorithms/data_tables/result_list.html"
+    search_fields = [
+        "job__creator__username",
+        "job__image__name",
+        "images__files__file",
+        "comment",
+    ]
+    columns = [
+        "job__created",
+        "job__creator__username",
+        "job__image__name",
+        "visibility",
+        "images__files__file",
+        "comment",
+    ]
+    order_by = "job__created"
+
+    def get_row_context(self, result, *args, **kwargs):
+        checker = ObjectPermissionChecker(self.request.user)
```

Comment:
How many queries does this result in? One for each row? Where has `prefetch_perms` gone?

Reply:
Good point, I did not check the queries, I'll have a look with the django debug toolbar.

Reply:
Added it back in,